yb-voyager/cmd/assessMigrationCommandBulk.go

```diff
@@ -0,0 +1,311 @@
+	Short: "Bulk Assessment of multiple schemas across one or more Oracle database instances",
+	Long:  "Bulk Assessment of multiple schemas across one or more Oracle database instances",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		assessMigrationBulk()
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(assessMigrationBulkCmd)
+
+	// defining flags
+	assessMigrationBulkCmd.Flags().StringVar(&fleetConfigPath, "fleet-config-file", "", "File containing the connection params for schema(s) to be assessed (required)")
+	BoolVar(assessMigrationBulkCmd.Flags(), &continueOnError, "continue-on-error", true, "If true, it will print the error message on console and continue to next schema’s assessment")
+	assessMigrationBulkCmd.Flags().Bool("ignore-exists", true, "If true, skip assessment if for a schema it's already done")
+	assessMigrationBulkCmd.Flags().StringVar(&bulkAssessmentDir, "bulk-assessment-dir", "", "Top-level directory storing the export-dir of each schema (default: pwd)")
+	BoolVar(assessMigrationBulkCmd.Flags(), &startClean, "start-clean", false, "Cleans up all the export-dirs in bulk assessment directory to start everything from scratch")
+
+	// marking mandatory flags
+	assessMigrationBulkCmd.MarkFlagRequired("fleet-config-file")
+	assessMigrationBulkCmd.MarkFlagRequired("bulk-assessment-dir")
+}
+
+func assessMigrationBulk() {
+	if startClean {
+		proceed := utils.AskPrompt(
+			"CAUTION: Using --start-clean will delete all progress in each export directory present inside the bulk-assessment-dir. " +
+				"Do you want to proceed")
+		if !proceed {
+			return
+		}
+
+		// cleaning all export-dir present inside bulk-assessment-dir
+		matches, err := filepath.Glob(fmt.Sprintf("%s/*-export-dir", bulkAssessmentDir))
+		if err != nil {
+			utils.ErrExit("error while cleaning up export directories: %s", err)
+		}
+		for _, match := range matches {
+			utils.CleanDir(match)
+		}
+
+		err = os.RemoveAll(filepath.Join(bulkAssessmentDir, "bulkAssessmentReport.html"))
+		if err != nil {
+			utils.ErrExit("failed to remove bulk assessment report: %s", err)
+		}
+		utils.CleanDir(filepath.Join(bulkAssessmentDir, "logs"))
+	}
+
+	dbConfigs, err := parseFleetConfigFile(fleetConfigPath)
+	if err != nil {
+		utils.ErrExit("failed to parse fleet config file: %v", err)
+	}
+
+	for _, dbConfig := range dbConfigs {
+		utils.PrintAndLog("\nAssessing '%s' schema\n", dbConfig.Schema)
+		err = executeAssessment(dbConfig)
+		if err != nil {
+			log.Errorf("failed to assess migration for schema %s: %v", dbConfig.Schema, err)
+			fmt.Printf("failed to assess migration for schema %s: %v\n", dbConfig.Schema, err)
+			if !continueOnError {
+				break
+			}
+		}
+	}
+
+	err = generateBulkAssessmentReport(dbConfigs)
+	if err != nil {
+		utils.ErrExit("failed to generate bulk assessment report: %s", err)
+	}
+}
+
+func executeAssessment(dbConfig AssessMigrationDBConfig) error {
+	log.Infof("executing assessment for schema %q", dbConfig.Schema)
```

Comment:
I would have imagined we could do this without having to create a new process. 
Can you explore if we can do that? 

One positive of that is that we don't have to deal with process handling (ensuring child is killed, different environments, etc. 

One negative is: we probably can't run all of them in parallel(if we ever need to) because there are a lot of global variables.

Reply:
Yes @makalaaneesh, initially i tried without a new process, using assessMigrationCmd.Run() and PreRun() functions internally.

Couple of challenges i faced because of which i chose to go with this approach(simpler):
1. We need a separate/independent migrationUUID and metaDB for each migration
2. Conflicts due to different directory structure for this command, got complicated to handle that in code.
     - When i was calling assessMigratioCmd.Run() and PreRun() from assessMigrationBulkCmd.go then it was not calling Root.PreRun() by default, which is mandatory for all the checks to happen and i believe that is because Root.PreRun() is called once which was happening for bulkCmd but not subsequent assessMig commands
     - assessMigrationCmd was not properly picking up the flags, which i was passing to its Run(cmd, args) as arguments.

In this some might be just solved by proper implementation but i think 1 and 2.1 points are still a bigger problem.

So using a new process approach seemed simpler and better to me. Let me know what do you think

Reply:
Yeah, i think our code in cmd package might not be very friendly to re-use easily. Makes sense I guess. 
Let's just discuss this implementation  with the whole team once in stand-up to see if there are other opinions.